lib/ace/editor.js

```diff
@@ -535,8 +535,26 @@ var Editor = function(renderer, session) {
             } else if (session.$mode.getMatching) {
                 var range = session.$mode.getMatching(self.session);
             }
-            if (range)
+            if (range) {
                 session.$bracketHighlight = session.addMarker(range, "ace_bracket", "text");
+            } else {
+                pos = session.findSurroundingBrackets(self.getCursorPosition());
+                if (!pos.mismatch) {
+                    if (pos) {
+                        if (pos.left && pos.right) {
```

Comment:
besides, you have a guarantee that mismatch == false => left and right are defined